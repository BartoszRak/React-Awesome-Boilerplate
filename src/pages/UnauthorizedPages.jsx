import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import Login from './Login'

export function UnauthorizedPages() {
  return (
    <Switch>
      <Route component={Login} path="/" />
      <Redirect from="*" to="/" />
    </Switch>
  )
}

export default UnauthorizedPages
