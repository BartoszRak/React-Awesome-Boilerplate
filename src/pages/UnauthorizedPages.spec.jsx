import React from 'react'
import { shallow } from 'enzyme'

import { UnauthorizedPages } from './UnauthorizedPages'

jest.mock('./Login', () => 'LoginMock')

const initComponent = overrides => {
  const mockProps = {}
  const mockMethods = {}
  const wrapper = shallow(<UnauthorizedPages {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('UnauthorizedPages: ', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
