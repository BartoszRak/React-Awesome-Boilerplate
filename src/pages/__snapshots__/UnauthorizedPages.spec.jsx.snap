// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthorizedPages:  should render as expected 1`] = `
<SwitchMock>
  <RouteMock
    component="LoginMock"
    path="/"
  />
  <RedirectMock
    from="*"
    to="/"
  />
</SwitchMock>
`;
